File: Best Time to Buy and Sell Stock.dart
class Solution {
  int maxProfit(List<int> prices) {
    if (prices.isEmpty) return 0;
    
    int minPrice = prices[0]; 
    int profit = 0;

    for (int price in prices) {
      if (price < minPrice) {
        minPrice = price;
      }
      else {
        profit = profit > price - minPrice ? profit : price - minPrice;
      }
    }
    return profit;
  }
}

File: etc.../Go etc.../Best Time to Buy and Sell Stock.go
package main

func maxProfit(prices []int) int {
	profit := 0
	minProfit := prices[0]

	for i := 1; i < len(prices); i++ {
		if prices[i] < minProfit {
			minProfit = prices[i]
		} else if (prices[i] - minProfit) > profit {
			profit = prices[i] - minProfit
		}
	}
	return profit
}
